app/app/static/custom/ourscript.js

```diff
@@ -342,3 +342,43 @@ window.onload = async (event) => {
 	rerenderTables();
 	
 };
+
+$('#dateSubmit').on('click', function(e) {
+	const startDate = document.getElementById("startDate").value;
+	const endDate = document.getElementById("endDate").value;
+	let startTime = startDate + " 00:00:00.0"
+	let endTime = endDate + " 23:59:59.9"
+
+	requestCSV({
+		startTime,
+		endTime
+	})
+})
+
+const requestCSV = async (data) => {
+	// This function requests a csv from the application with the data
+	// between the two specified dates
+
+	try{
+		const response = await fetch("CSV", {
+			method: "POST",
+			body: JSON.stringify(data),
+			headers: {
+				'Content-Type': 'application/json'
+			},
+		})
+		let filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
+		let filename = filenameRegex.exec(response.headers.get('Content-Disposition'))[0].replace('filename=','');
+		const dataResponse = await response.blob();
+		let downloadURL = URL.createObjectURL(dataResponse);
+		let a = document.createElement("a");
```

Comment:
You never dereferenced the variable so you can just use `const` instead of `let`

Reply:
Good to know!